pkg/reconciler/common/transformers.go

```diff
@@ -874,3 +874,36 @@ func ReplaceNamespaceInClusterRoleBinding(targetNamespace string) mf.Transformer
 		return nil
 	}
 }
+
+// updates "metadata.namespace" and under "spec"
+// TODO: we have different transformer for each kind
+// TODO: replaces all the existing transformers(used to update namespace) with this.
+func ReplaceNamespace(newNamespace string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		switch u.GetKind() {
+		case "Deployment", "Pod", "Service", "ConfigMap", "Secret", "ServiceAccount":
```

Comment:
may we keep this default, others we add base on their type and spec

Reply:
@piyush-garg do not want to patch the resource not known. Hence applying to known kind only.